src/main/java/edu/iu/terracotta/service/app/impl/AssessmentServiceImpl.java

```diff
@@ -430,4 +477,118 @@ public AssessmentDto duplicateAssessment(long assessmentId, long treatmentId) th
+
+        if (!exposureGroupCondition.isPresent()){
+            throw new AssessmentNotMatchingException("Error 130: This assignment does not have a condition assigned for the participant group.");
+        }
+
+        return retrieveTreatmentAssessment(exposureGroupCondition.get().getCondition().getConditionId(), assignment.getAssignmentId());
+    }
+
+    @Override
+    public Assessment getAssessmentByConditionId(Long experimentId, String canvasAssignmentId, Long conditionId) throws AssessmentNotMatchingException {
+        Assignment assignment = allRepositories.assignmentRepository.findByExposure_Experiment_ExperimentIdAndLmsAssignmentId(experimentId, canvasAssignmentId);
+
+        if (assignment == null) {
+            throw new AssessmentNotMatchingException("Error 127: This assignment does not exist in Terracotta for this experiment");
+        }
+
+        return retrieveTreatmentAssessment(conditionId, assignment.getAssignmentId());
+    }
+
+    private Assessment retrieveTreatmentAssessment(long conditionId, long assignmentId) throws AssessmentNotMatchingException {
+        List<Treatment> treatments = allRepositories.treatmentRepository.findByCondition_ConditionIdAndAssignment_AssignmentId(conditionId, assignmentId);
+
+        if (treatments.isEmpty()) {
+            throw new AssessmentNotMatchingException("Error 131: This assignment does not have a treatment assigned.");
+        }
+
+        if (treatments.size() > 1) {
+            throw new AssessmentNotMatchingException("Error 132: This assignment has ambiguous treatments. Please contact a Terracotta administrator");
+        }
+
+        if (treatments.get(0).getAssessment() == null) {
+            throw new AssessmentNotMatchingException("Error 133: The treatment for this assignment does not have an assessment created");
+        }
+
+        return treatments.get(0).getAssessment();
+    }
+
+    @Override
+    public AssessmentDto viewAssessment(long experimentId, SecuredInfo securedInfo)
+            throws ExperimentNotMatchingException, ParticipantNotMatchingException, AssessmentNotMatchingException,
+                GroupNotMatchingException, ParticipantNotUpdatedException, AssignmentNotMatchingException {
+        Optional<Experiment> experiment = allRepositories.experimentRepository.findById(experimentId);
+
+        if (!experiment.isPresent()) {
+            throw new ExperimentNotMatchingException(TextConstants.EXPERIMENT_NOT_MATCHING);
+        }
+
+        Participant participant = allRepositories.participantRepository.findByExperiment_ExperimentIdAndLtiUserEntity_UserKey(experimentId, securedInfo.getUserId());
+
+        if (participant == null) {
+            throw new ParticipantNotMatchingException(TextConstants.PARTICIPANT_NOT_MATCHING);
+        }
```

Comment:
I like this refactor. We can just remove these lines now. The code should refresh the participants before it can determine there is no matching participant, and handleExperimentParticipant handles that now.

Reply:
Good catch! I'll remove those lines now.